### PR TITLE
feat: local inference

### PR DESCRIPTION
**Pull request by qued**
Splits `partition_pdf` into two paths, one used for local inference when `url` is `None`, another for inference via api when `url` is a string. For now, there are special accommodations for the fact that the api selects a model type based on a form field, but we intend to move to using the url route to pick a model. Where there is special code for this, it's noted.

Added dependencies and make targets for `unstructured-inference`.

#### Testing:
- Install new dependencies with `make install-local-inference` (might be best to install in a new environment)
- Run this code from root directory:
```python
from unstructured.partition.pdf import partition_pdf, _partition_pdf_via_local

partition_pdf(filename="example-docs/layout-parser-paper-fast.pdf", url=None)
_partition_pdf_via_local(filename="example-docs/layout-parser-paper-fast.pdf")

```

### CODE REVIEW

File: unstructured/partition/pdf.py [hunk 1]
```diff
@@ -47,10 +68,30 @@ def partition_pdf(
...
         raise ValueError(f"response status code = {response.status_code}")
+
+
+def _partition_pdf_via_local(
+    filename: str = "",
+    file: Optional[bytes] = None,
+    template: Optional[str] = None,
+) -> List[Element]:
+    """Partition using package installed locally."""
+    from unstructured_inference.inference.layout import (
```
Comment (MthwRobinson) [inline unstructured/partition/pdf.py:87]: If this results in a `ModuleNotFoundError` or `ImportError` (I forget which one of those is more generally), should we raise an error with a message that tells the user to install the `local-inference` deps?

File: requirements/local-inference.in [hunk 2]
```diff
@@ -0,0 +1 @@
+unstructured-inference>=0.2.1
```
Comment (MthwRobinson) [inline requirements/local-inference.in:1]: Do we want to put this as an optional requirement in `setup.py` so you can do `pip install unstructured[local_inference]`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ## 0.3.5-dev6
```
Comment (MthwRobinson) [inline CHANGELOG.md:1]: With local inference in there now, we can probably release a `0.3.6`

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
One error handling suggestion and a question on how we want to handle the optional dependencies.

File: requirements/local-inference.in (hunk 2 above)
Comment (qued) [reply requirements/local-inference.in:1]: Yep. Added in [8b6d379](https://github.com/Unstructured-IO/unstructured/pull/125/commits/8b6d379b215310801ccc516f47945f3670d71a51).

File: unstructured/partition/pdf.py (hunk 1 above)
Comment (qued) [reply unstructured/partition/pdf.py]: Added error handling with instructions in [d5445d9](https://github.com/Unstructured-IO/unstructured/pull/125/commits/d5445d9661da22ad573229c57c3ef480544e94d4).

File: CHANGELOG.md (hunk 3 above)
Comment (qued) [reply CHANGELOG.md]: Changed version to release version in [4e65e6e](https://github.com/Unstructured-IO/unstructured/pull/125/commits/4e65e6e326e871b2e626af12075aacad8e684abe).

**Review by MthwRobinson — APPROVED** [review]
LGTM!

**Pull request merged**